yarn-project/noir-protocol-circuits/src/crates/types/src/abis/nullifier_key_validation_request.nr

```diff
@@ -0,0 +1,109 @@
+use dep::std::cmp::Eq;
+use crate::{
+    address::AztecAddress,
+    traits::Empty,
+    grumpkin_point::GrumpkinPoint,
+    grumpkin_private_key::GrumpkinPrivateKey,
+};
+
+global NULLIFIER_KEY_VALIDATION_REQUEST_SERIALIZED_LEN = 4;
+global NULLIFIER_KEY_VALIDATION_REQUEST_CONTEXT_SERIALIZED_LEN = 5;
+
+struct NullifierKeyValidationRequest {
+    public_key: GrumpkinPoint,
+    secret_key: GrumpkinPrivateKey,
+}
+
+impl Eq for NullifierKeyValidationRequest {
+    fn eq(self, request: NullifierKeyValidationRequest) -> bool {
+        (request.public_key.eq(self.public_key))
+            & (request.secret_key.eq(self.secret_key))
+    }
+}
+
+impl Empty for NullifierKeyValidationRequest {
+    fn empty() -> Self {
+        NullifierKeyValidationRequest {
+            public_key: GrumpkinPoint::zero(),
+            secret_key: GrumpkinPrivateKey::zero(),
+        }
+    }
+}
+
+impl NullifierKeyValidationRequest {
+    pub fn is_empty(self) -> bool {
+        self.public_key.is_zero() & self.secret_key.is_zero()
+    }
+
+    pub fn serialize(self) -> [Field; NULLIFIER_KEY_VALIDATION_REQUEST_SERIALIZED_LEN] {
+        [
+            self.public_key.x,
+            self.public_key.y,
+            self.secret_key.high,
+            self.secret_key.low,
+        ]
+    }
+
+    pub fn deserialise(fields: [Field; NULLIFIER_KEY_VALIDATION_REQUEST_SERIALIZED_LEN]) -> Self {
+        Self {
+            public_key: GrumpkinPoint::new(fields[0], fields[1]),
+            secret_key: GrumpkinPrivateKey::new(fields[2], fields[3]),
+        }
+    }
+
+    pub fn to_context(self, contract_address: AztecAddress) -> NullifierKeyValidationRequestContext {
+        NullifierKeyValidationRequestContext {
+            public_key: self.public_key,
+            secret_key: self.secret_key,
+            contract_address,
```

Comment:
In the spec, I called these fields `claimed_parent_public_key`, `hardened_child_secret_key`. 

The names `public_key` and `secret_key` give the impression that these are a keypair, but they're not. The word `claimed` could be omitted, I suppose. But the validation request is technically validating that this claimed public key is the parent of the given hardened child secret key.
I think verbose names are quite important for unusual concepts such as this - or at least names that match up against a spec, to help with auditing.